src/ngraph/runtime/dynamic_wrapper/dynamic_wrapper_backend.hpp

```diff
@@ -0,0 +1,112 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "ngraph/runtime/backend.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/runtime/tensor.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace dynamic_wrapper
+        {
+            class DynamicWrapperBackend;
+            class WrappedExecutable;
+            class WrappedStaticTensor;
+            class WrappedDynamicTensor;
+        }
+    }
+}
+
+class ngraph::runtime::dynamic_wrapper::DynamicWrapperBackend : public Backend
```

Comment:
Can you add some documentation about the purpose of each wrapper class ? For example, WrappedExecutable enables late compilation, code-caching, .. etc. 
I also not so fond of the class names. Maybe something like DynamicBackend, DynamicExecutable would be better.

Reply:
Will add some documentation.

For the class names, I kind of like the idea of having `Wrapper` somewhere in the name because it emphasizes that dynamic shapes are not being implemented "directly". It doesn't overpromise, if you will. :) Let me think about it.

Reply:
Decided to go with `DynamicBackend`, `DynamicExecutable`, `DynamicTensor`.